### ISSUE TITLE
Implement better handling of private routes

### ISSUE THREAD

**Issue by pattruong** [brewbean/client#23]
- ~~create a header container with auth knowledge~~
  - ~~Don’t use “loading...” in user context but pass to container view to render special loading display or 401~~
  - Ended up being handled well by `<AuthRoute />` wrapper
- ~~Add array representation for mapping routes~~
  - Ended up being buggy with rendering different options and layout. Ultimately could load but had some re-rendering steps that made page load look bad. Not worth the tradeoffs
- [x] if a user has logged in give them the benefit of doubt and don’t load 401 on private route page load
  - Basically `isAuthenticated || needRefresh` (optimistic)
  - Added `PENDING` status check as well (User Context now init w/ `PENDING` status)
- [x] Add `authOnlyPaths` handling for param routes: `/profile/:id/:entry` (use `matchPath`)
